src/backend/distributed/planner/query_pushdown_planning.c

```diff
@@ -281,21 +281,17 @@ JoinTreeContainsSubqueryWalker(Node *joinTreeNode, void *context)
 bool
 WhereOrHavingClauseContainsSubquery(Query *query)
 {
-	FromExpr *joinTree = query->jointree;
-
 	if (FindNodeCheck(query->havingQual, IsNodeSubquery))
 	{
 		return true;
 	}
 
-	if (!joinTree)
+	if (!query->jointree)
 	{
 		return false;
 	}
 
-	Node *queryQuals = joinTree->quals;
-
-	return FindNodeCheck(queryQuals, IsNodeSubquery);
+	return FindNodeCheck((Node *) query->jointree, IsNodeSubquery);
```

Comment:
Would be good to add a comment on why we have jointree instead of just the quals.